### PR TITLE
Work around test discovery crash when running on older OS versions due to unguarded use of .timeLimit

### PR DESCRIPTION
**Pull request by stmontgomery**
Fix a test discovery crash when running on older Apple OS versions which don't support `Clock` and related APIs.

### Checklist:

- [x] Code and documentation should follow the style of the [Style Guide](https://github.com/apple/swift-testing/blob/main/Documentation/StyleGuide.md).
- [x] If public symbols are renamed or modified, DocC references should be updated.

### CODE REVIEW

**Comment by stmontgomery** [comment]
@swift-ci please test

**Comment by stmontgomery** [comment]
@swift-ci please test

File: Tests/TestingTests/Traits/TimeLimitTraitTests.swift [hunk 1]
```diff
@@ -243,17 +243,21 @@ struct TimeLimitTraitTests {
...
 
-func timeLimitIfAvailable(minutes: UInt64) -> any SuiteTrait {
+private func _timeLimitIfAvailable(minutes: UInt64) -> any SuiteTrait {
   // @available can't be applied to a suite type, so we can't mark the suite as
-  // available only on newer OSes.
+  // available only on newer OSes. In addition, there is a related, known bug
+  // where traits with conditional API availability are not guarded by
+  // `@available` attributes on their associated `@Test` function
+  // (rdar://127811571). That is not directly relevant here but is worth noting
+  // if this trait is ever applied to `@Test` functions in this file.
```
Comment (stmontgomery) [inline Tests/TestingTests/Traits/TimeLimitTraitTests.swift:252]: No functional change in this file, just noting this related bug

**Review by grynspan — APPROVED** [review]

**Comment by stmontgomery** [comment]
@swift-ci please test Windows

**Pull request merged**